app/controllers/withdraw_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class WithdrawController < ApplicationController
+  before_action :user_has_permissions?, only: :new
+
+  def create
+    @document = Document.with_current_edition.find_by_param(params[:id])
+    @public_explanation = @document.current_edition.status.details&.public_explanation || params[:public_explanation]
+    if !current_user.has_permission?(User::PRE_RELEASE_FEATURES_PERMISSION)
+      render :withdraw
+      return
+    end
+
+    if current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION)
+      render :create
+    else
+      render :non_managing_editor
+    end
+  end
+
+  def new
+    document = Document.with_current_edition.find_by_param(params[:id])
+    public_explanation = params[:public_explanation]
+    issues = Requirements::PublicExplanationChecker.new(public_explanation).pre_withdrawal_issues
```

Comment:
Ideally we should name the requirements checkers after the feature, so it's clear where they're being used e.g. 'UnwithdrawChecker' or something.

Reply:
Change to `WithdrawalChecker`